src/main/java/us/ihmc/handsros2/ezGripper/EZGripper.java

```diff
@@ -0,0 +1,475 @@
+ *
+ * This version also incorporates the higher-level management logic such as operation modes,
+ * calibration status, and automatic cooldown temperature limits.
+ */
+public class EZGripper implements HandInterface
+{
+   public static final int RAW_RANGE_OF_MOTION = 2500;
+
+   public enum OperationMode
+   {
+      POSITION_CONTROL, CALIBRATION, ERROR_RESET, COOLDOWN;
+
+      public static final OperationMode[] values = values();
+
+      public static OperationMode fromByte(byte ordinal)
+      {
+         return values[ordinal];
+      }
+
+      public byte toByte()
+      {
+         return (byte) this.ordinal();
+      }
+   }
+
+   public static final int DISABLE_AUTO_COOLDOWN = 255;
+
+   private final String identifier;
+   private final RobotSide robotSide;
+
+   private final YoDouble goalPosition;
+   private final YoDouble maxEffort;
+   private final YoBoolean torqueOn;
+
+   private final YoDouble currentPosition;
+   private final YoDouble currentEffort;
+   private final YoInteger temperature;
+   private final YoInteger realtimeTick;
+   private final YoInteger errorCode;
+
+   private final YoBoolean isCalibrated;
+   private final YoDouble temperatureLimit;
+   private final YoRegistry registry;
+
+   private final YoEnum<OperationMode> desiredOperationMode;
+   private final YoEnum<OperationMode> operationMode;
+   private final StateMachine<OperationMode, State> stateMachine;
+
+   public EZGripper(String identifier, RobotSide robotSide)
+   {
+      this(new YoRegistry("EZGripper_" + identifier + "_" + robotSide.name()), identifier, robotSide);
+   }
+
+   public EZGripper(YoRegistry registry, String identifier, RobotSide robotSide)
+   {
+      this.identifier = identifier;
+      this.robotSide = robotSide;
+      this.registry = registry;
+
+      String prefix = robotSide.name() + "EZGripper";
+
+      // Low level control
+      goalPosition = new YoDouble(prefix + "GoalPosition", registry);
+      maxEffort = new YoDouble(prefix + "MaxEffort", registry);
+      torqueOn = new YoBoolean(prefix + "TorqueOn", registry);
+
+      // State
+      currentPosition = new YoDouble(prefix + "CurrentPosition", registry);
+      currentEffort = new YoDouble(prefix + "CurrentEffort", registry);
+      temperature = new YoInteger(prefix + "Temperature", registry);
+      realtimeTick = new YoInteger(prefix + "RealtimeTick", registry);
+      errorCode = new YoInteger(prefix + "ErrorCode", registry);
+
+      // Manager state
+      isCalibrated = new YoBoolean(prefix + "IsCalibrated", registry);
+      isCalibrated.set(false);
+      temperatureLimit = new YoDouble(prefix + "TemperatureLimit", registry);
+      temperatureLimit.set(DISABLE_AUTO_COOLDOWN);
+
+      desiredOperationMode = new YoEnum<>(prefix + "DesiredOperationMode", registry, OperationMode.class);
+      desiredOperationMode.set(OperationMode.POSITION_CONTROL);
+      operationMode = new YoEnum<>(prefix + "operationMode", registry, OperationMode.class);
```

Comment:
Just for consistency
```suggestion
      operationMode = new YoEnum<>(prefix + "OperationMode", registry, OperationMode.class);
```